Core-perf: Netty HttpClient and Ok HttpClient performance (#27982)
Azure/azure-sdk-for-java: sdk/core/azure-core-http-okhttp/CHANGELOG.md

## 1.9.0-beta.1 (Unreleased)

### Features Added

> - The `OkHttpAsyncClientProvider.createInstance()` now have option to share one single shared `HttpClient`.

Reviewer: ```suggestion
- The `OkHttpAsyncClientProvider.createInstance()` now has the option to share a shared `HttpClient`.
```

>   A new configuration environment variable `AZURE_ENABLE_HTTP_CLIENT_SHARING` has been added for backwards
>   compatibility. Setting it to `true` before starting the process to use single shared `HttpClient`.

Reviewer: Since the flag has changed from opt-out to opt-in this statement can be changed.

```suggestion
  Set the environment variable `AZURE_ENABLE_HTTP_CLIENT_SHARING`  to `true` before starting the process to use the shared `HttpClient`.
```
